File: commonlib/src/main/java/com/akame/commonlib/utils/ActivityCollector.java
package com.akame.commonlib.utils;

import android.app.Activity;

import java.util.HashSet;
import java.util.Set;

/**
 * @Author: Restring
 * @Date: 2018/4/23
 * @Description:
 */
public class ActivityCollector {
    private static ActivityCollector activityCollector;

    public synchronized static ActivityCollector getInstance() {
        if (activityCollector == null) {
            activityCollector = new ActivityCollector();
        }
        return activityCollector;
    }

    private Set<Activity> allActivities;

    public void addActivity(Activity act) {
        if (allActivities == null) {
            allActivities = new HashSet<>();
        }
        allActivities.add(act);
    }

    public void removeActivity(Activity act) {
        if (allActivities != null) {
            allActivities.remove(act);
        }
    }

    public void removeAllActivity() {
        if (allActivities != null) {
            synchronized (allActivities) {
                for (Activity act : allActivities) {
                    act.finish();
                }
            }
        }
    }

    public void exitApp() {
        removeAllActivity();
        android.os.Process.killProcess(android.os.Process.myPid());
        System.exit(0);
    }

}

File: developkit/src/main/java/com/akame/developkit/util/ActivityCollector.kt
package com.akame.developkit.util

import android.app.Activity
import kotlin.system.exitProcess

object ActivityCollector {
    private val activitySet = HashSet<Activity>()

    fun addActivity(activity: Activity) {
        activitySet.add(activity)
    }

    fun removeActivity(activity: Activity) {
        activitySet.remove(activity)
    }

    fun removeAllActivity() {
        synchronized(activitySet) {
            activitySet.forEach {
                it.finish()
            }
        }
    }

    fun exitApp() {
        removeAllActivity()
        android.os.Process.killProcess(android.os.Process.myPid())
        exitProcess(0)
    }
}